Add an S3 Gateway Endpoint to the VPC

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR adds an [S3 gateway endpoint](https://docs.aws.amazon.com/vpc/latest/privatelink/vpce-gateway.html) in the VPC for SharedServices. This is part of the work for https://github.com/cisagov/cool-system/issues/197.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

This improves security for any communications with the S3 service from anything using the SharedServices VPC.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
This functionality was tested as part of the testing for https://github.com/cisagov/cool-images-assessment-images/pull/4.
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.

mcdonnnj committed:
Add AWS S3 Gateway endpoint

Add a vpc_endpoints.tf file to house any endpoints for this project. Add an S3
gateway endpoint for this project in the aforementioned file.

mcdonnnj committed:
Associate the S3 endpoint

Associate the S3 endpoint with the public route table and each of the private
route tables.

mcdonnnj committed:
Add the S3 Gateway endpoint to the outputs

This will allow other Terraform projects to access information about the S3
Gateway endpoint when using the remote state for this project.

dav3r commented on outputs.tf:
```diff
@@ -87,3 +87,8 @@ output "vpc_dhcp_options_association" {
+
+output "vpc_endpoint_s3" {
+  value       = aws_vpc_endpoint.s3
+  description = "The S3 Gateway endpoint for the Shared Services VPC."
```
Minor thang, but I agree with keeping "gateway" lower case (as you did in your comments).
```suggestion
  description = "The S3 gateway endpoint for the Shared Services VPC."
```

dav3r reviewed (approved):
Nice job- this LGTM! 🔚 👉

jsf9k reviewed (approved):
Nice work!

mcdonnnj committed:
Update variable description

Update capitalization to be consistent with how it is used elsewhere in the project.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
